pettingzoo/mpe/scenarios/group_spread.py

```diff
@@ -150,34 +142,26 @@ def global_reward(self, world):
         return rew
 
     def observation(self, agent, world):
+        # own position, velocity, color
+        color = [np.eye(self.num_colors)[agent.color_idx]] if self.obs_onehot_colors else [agent.color]
+        entity_features = [agent.state.p_pos] + [agent.state.p_vel] + color
```

Comment:
This substitutes the `[agent.state.p_vel] + [agent.state.p_pos] + [np.eye(self.num_colors)[agent.group]]` in the last line of this function before (wanted to make these 3 blocks as separate code).

Also, here I think there are 2 actually meaningful bugs to do with colors of agents:

1. Previously, the color of the agent itself was extracted as `np.eye(self.num_colors)[agent.group]` but I think this should be `np.eye(self.num_colors)[agent.color_idx]` as used now for one-hot colors. At the beginning of each episode, `color_idxs` are randomised (shuffled for each group), so the index access previously was to the wrong color!

2. The previous code always gave the color of the agent as the one-hot array. However, previously the colors of the landmarks were provided as real color values and not one-hots whenever `randomise_all_colors=True` in which case it was impossible for agents to identify which landmark matches their color.

Reply:
Yes these are both bugs, I think #1 probably explains the performance issues with 3 agents!

Reply:
Further simplified color obs generation --> now all from color config defined above

Reply:
Great finds, especially 1., will start some runs to check the shuffling problem after we merge this!